Fix group diagram scroll behaviour

vorbrodt opened the pull request:
Signed-off-by: Maximilian Vorbrodt maximilian.vorbrodt@hotmail.com

## Hey, I just made a Pull Request!

<!-- Please describe what you added, and add a screenshot if possible.
     That makes it easier to understand the change so we can :shipit: faster. -->

This is a fix related to issue #11429.

It adds a background to the the group diagram to make it clearer where the actual graph is located. It also makes it so that the the graph fills up the available space which is visible on the page. What this means is that the user will never have to scroll up or down on the page as the full graph will always be in view.  

Here is a short demonstration of how it looks:

https://user-images.githubusercontent.com/64839037/201689375-dfc2e3fd-457b-41cc-957e-eaab03a3429a.mp4


#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Rugvip commented on plugins/explore/src/components/GroupsExplorerContent/GroupsDiagram.tsx:
```diff
@@ -46,6 +46,12 @@ const useStyles = makeStyles((theme: BackstageTheme) => ({
     flex: 1,
     minHeight: 0,
   },
+  graphWrapper: {
+    position: 'relative',
+    flex: 1,
+    minHeight: 0,
+    display: 'flex',
+  },
```
Do you see a way to get this to fill up the rest of the empty space on the page without using any magic numbers? I feel there's enough vertical space that the best solution to this problem would be if the graph simply fills out the rest of the page and there's no vertical scrolling at all. What do you think?

vorbrodt replied:
I think the current solution works pretty well actually, but if you think that would be a better solution I will look into it. 👍 

If we are going with that solution I guess we will be going back to the previous scroll behaviour, otherwise it will be difficult for the user to scroll down. I will also remove the added background to the graph, so it doesn't blend together with the sidebar as it fills the empty space.

Rugvip replied:
I haven't checked, but does this solve the original issue of not being able to scroll back up if the mouse is hovered above the diagram?

What I'm proposing is the make the graph fill the page in a way where it is not possible to scroll at all

vorbrodt replied:
Yeah, I see. When reading what people were saying it seemed that the underlying issue was that it's unexpected behaviour when you can scroll down when hovering the graph, but not scroll up. What the current solution aims to do is to make it clear that you should not expect to scroll up or down when hovering the graph (like google maps for instance),  and it does this by adding a background to the graph and removing scroll behaviour when hovering the graph.

vorbrodt commented:
Care to take another look at the new solution? 😄  @Rugvip 

I added a new desc for this PR and added a demo of how it looks with the new solution. I think the new solution works great, as the graph now fills all the available and visible space making it so the user will never have to scroll up and down, but still always can see the entire graph.

Rugvip commented on packages/core-components/src/components/DependencyGraph/DependencyGraph.tsx:
```diff
@@ -394,7 +395,7 @@ export function DependencyGraph<NodeData, EdgeData>(
       ref={containerRef}
       {...svgProps}
       width="100%"
-      height={maxHeight}
+      height={minHeight}
```
How does this affect all other usages of this component? We may need to surface this through a prop if needed, for example something like `fit?: 'grow' | 'contain'`.

Rugvip commented on .changeset/lucky-carrots-stare.md:
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/core-components': patch
+'@backstage/plugin-explore': patch
```
We'll need to split this into two separate changesets. The text here doesn't make sense for the `core-components` changes right now

Rugvip commented on plugins/explore/src/components/GroupsExplorerContent/GroupsDiagram.tsx:
```diff
@@ -222,21 +237,26 @@ export function GroupsDiagram() {
-        <ZoomOutMap style={{ verticalAlign: 'bottom' }} /> Use pinch &amp; zoom
-        to move around the diagram.
-      </Typography>
+      <Paper className={classes.graphWrapper}>
```
Do you think the paper here is still needed? If worried it might just limit the size of the graph without any tangible benefit

Rugvip reviewed (changes requested):
Nice! 👍 I think we're pretty much good, just a couple of nits and we'll need to figure out how to package the component update.

Rugvip reviewed (approved):
Nice, thank you! 👍 😁 

Let's :shipit: 🎉
